org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/preferences/BuilderConfigurationBlock.java

```diff
@@ -52,6 +55,8 @@
 public class BuilderConfigurationBlock extends OptionsConfigurationBlock {
 	
 	protected static final String[] BOOLEAN_VALUES = new String[] { IPreferenceStore.TRUE, IPreferenceStore.FALSE };
+
+	private static final Predicate<Entry<String, ValueDifference<String>>> AFFECTS_BUILDER_PREDICATE = new AffectsBuilderPredicate();
```

Comment:
I'd rather like to inject this one. Can we use a dedicated type here, though?

Reply:
Maybe a type with a method `isAffectingBuild(Map<String, ValueDiff<String>>)` would be easier to grasp and allow for easier customizing, too?